server/src/main/java/org/opensearch/gateway/remote/model/AbstractRemoteBlobStore.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.model;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.PATH_DELIMITER;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.gateway.remote.RemoteClusterStateUtils;
+import org.opensearch.index.translog.transfer.BlobStoreTransferService;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * Abstract class for a blob type storage
+ *
+ * @param <T> The entity which can be uploaded to / downloaded from blob store
+ * @param <U> The concrete class implementing {@link RemoteObject} which is used as a wrapper for T entity.
+ */
+public class AbstractRemoteBlobStore<T, U extends AbstractRemoteBlobObject<T>> implements RemoteObjectStore<T, U> {
```

Comment:
nit: Name is misleading there is nothing `abstract` here?

Reply:
Yes there are no abstract methods. (Also realized that the abstract class modifier is missing)

I wanted it abstract because we do not want to allow anyone to create an instance of this class.

Reply:
Sorry I am confused, there is nothing for the concrete classes to implement, why do not want to allow creation of the instance

Reply:
Yes, right there no abstract methods right now. I was thinking if we need to add abstract methods in the future, then this class would help. But we can skip that for now I guess.
I will rename the class to remove the Abstract from the name.